### PR TITLE
Rel 539546 i want sample usage of direct mode available on GitHub

### PR DESCRIPTION
**Pull request by BartoszWierzbaRelativity**

### CODE REVIEW

File: Relativity.Transfer.Sample/SampleHelperFactory.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    using System.ComponentModel;
+
+    using Relativity.Transfer.Sample.Enums;
+    using Relativity.Transfer.Sample.SampleHelpers;
+
+    public class SampleHelperFactory
+    {
+        public AbstractSampleHelper GetSampleHelper(TransferMode transferMode)
+        {
+            if (transferMode == TransferMode.Aspera)
```
Comment (KamilKalitaRel) [inline Relativity.Transfer.Sample/SampleHelperFactory.cs]: Minor: switch would be more concise.

File: Relativity.Transfer.Sample/SampleHelpers/AsperaSampleHelper.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+// // <copyright file="AsperaSampleHelper.cs" company="kCura Corp">
+// //   kCura Corp (C) 2021
+// // </copyright>
+// // ----------------------------------------------------------------------------
+
+namespace Relativity.Transfer.Sample.SampleHelpers
+{
+    public class AsperaSampleHelper : AbstractSampleHelper
+    {
+        public override string TransferModeName { get; } = "Aspera";
```
Comment (KamilKalitaRel) [inline Relativity.Transfer.Sample/SampleHelpers/AsperaSampleHelper.cs]: Maybe `TransferMode.Aspera.ToString()` could be used instead?

**Comment by KamilKalitaRel** [comment]
Excluding the code to a separate sample class is a good idea.

We should have a description how to use the Direct mode in README, with a clarification how it differs from Aspera.

File: Relativity.Transfer.Sample/SampleHelperFactory.cs (hunk 1 above)
Comment (BartoszWierzbaRelativity) [reply Relativity.Transfer.Sample/SampleHelperFactory.cs]: Done

File: Relativity.Transfer.Sample/SampleHelpers/AsperaSampleHelper.cs (hunk 2 above)
Comment (BartoszWierzbaRelativity) [reply Relativity.Transfer.Sample/SampleHelpers/AsperaSampleHelper.cs]: Done

File: README.md [hunk 3]
```diff
@@ -53,168 +56,30 @@ If a direct File Share transfer is used, there is effectively no limit to the pa
...
 As part of these updates, a GlobalSetting variable has been added to adjust the behavior when a path that is too long for the chosen client to transfer is found during enumeration. This setting, called `SkipTooLongPaths`, is a boolean value. If `true`, any paths longer than the client supported maximum will be classified as an Error Path, and won't be transferred. However, enumeration and the transfer of all other valid paths will complete as part of the transfer job. If `false`, the enumeration will throw a fatal PathTooLongException upon encountering an invalid path length, and the transfer will fail. No files will be transferred in this situation.
 
-## Sample solution and tutorial
-The `Sample.sln` solution is an out-of-the-box template for developing your own custom transfer applications and demonstrates Aspera API usage.
+## Sample solution
+The `Sample.sln` solution is an out-of-the-box template for developing your own custom transfer applications and demonstrates Aspera and Fileshare API usage.
 
 Prerequisites for running the solution:
 
 * Visual Studio 2015 or 2017
```
Comment (KamilKalitaRel) [inline README.md]: Can't be newer? ;)

File: README.md [hunk 4]
```diff
@@ -53,168 +56,30 @@ If a direct File Share transfer is used, there is effectively no limit to the pa
...
-      <dependentAssembly>
-        <assemblyIdentity name="Newtonsoft.Json" publicKeyToken="30ad4fe6b2a6aeed" culture="neutral" />
-        <bindingRedirect oldVersion="0.0.0.0-10.0.0.0" newVersion="6.0.0.0" />
-      </dependentAssembly>
-    </assemblyBinding>
-  </runtime>
-</configuration>
-```
+Ensure the following 5 settings found in app.config are updated with valid parameters:
+* TransferMode
```
Comment (KamilKalitaRel) [inline README.md:71]: Please add a note that it is either `Aspera` or `Direct`.

File: README.md [hunk 5]
```diff
@@ -1165,7 +337,7 @@ The API caller uses the transfer host to construct the appropriate transfer clie
 using (ITransferClient client = host.CreateClient(new AsperaClientConfiguration()))
 { }
 
-// I need a file share client.
+// I need a Fileshare client.
```
Comment (KamilKalitaRel) [inline README.md]: Sometimes it's spelled "file share", sometimes "Fileshare". Azure Portal uses "file share".

File: README.md [hunk 6]
```diff
@@ -110,21 +112,22 @@ The `CreateTransferLog()` method uses a [Relativity Logging](https://platform.re
...
+Client will always be your best transfer option. For more information, see [Dynamic Transfer Client](#dynamic-transfer-client) and [ITransferClient](#itransferclient).
 
 #### Subscribe to transfer events
 Since transfer events are used for both upload and download operations, the demo wraps this within the `CreateTransferContext()` method to construct the [TransferContext](#transfer-events-and-statistics) object and write event details to the console. This object is used to decouple the event logic of the transfer - for example, progress and  statistics - from the host and the client.
 
 ### Demo
-Real-world applications typically involve large or even massive datasets that not only require better transfer request management but provide real-time data rate, progress, and time remaining to their users.
+Real-world applications typically involve large or even massive datasets that not only require better transfer request management but provide real-time data rate, progress, and time remaining to their users.file share
```
Comment (KamilKalitaRel) [inline README.md]: Too much `.file share` ;)

**Review by KamilKalitaRel — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (BartoszWierzbaRelativity) [reply README.md]: Done

File: README.md (hunk 4 above)
Comment (BartoszWierzbaRelativity) [reply README.md:71]: Done

File: README.md (hunk 5 above)
Comment (BartoszWierzbaRelativity) [reply README.md]: Done

File: README.md (hunk 6 above)
Comment (BartoszWierzbaRelativity) [reply README.md]: Done

**Pull request merged**